tests/core/a-entity.test.js

```diff
@@ -98,6 +98,48 @@ suite('a-entity', function () {
     });
   });
 
+  suite('addState', function () {
+    test('add state', function () {
+      var el = this.el;
+      el.states = [];
+      el.addState('happy');
+      assert.ok(el.states[0] === 'happy');
+    });
+
+    test('it does not add an existing state', function () {
+      var el = this.el;
+      el.states = ['happy'];
+      el.addState('happy');
+      assert.ok(el.states.length === 1);
+    });
+  });
+
+  suite('removeState', function () {
+    test('removes existing state', function () {
```

Comment:
Can add cases for:

- Removing non-existent state.
- Removing one state when entity has multiple states.